splashkit/splashkit-translator
# fix: modernise ruby toolchain

This PR updates the Ruby toolchains to `2.7.5`.

- Set Ruby version to `2.7.5`
- Update all dependencies
- Set bundle version to v2 (v1 has already been deprecated)

## How to test

- Check out the branch
- Make sure to use a [Ruby Version Manager](https://github.com/rbenv/rbenv) to install Ruby 2.7.5 (do not use the one that comes with macOS)
- Run `bundle install`
- Verify the translator run successfully
  ```sh
  ./translate -i ../splashkit-core -o ~/Desktop/translated -g PYTHON
  ```
- This takes a while (~ 45.67s on my computer)
- Verify the file content
  ```sh
  ❯ wc -l ~/Desktop/translated/python/splashkit.py
  8904 /Users/tanle/Desktop/translated/python/splashkit.py
  ```
Tested on macOS 12.3.1 (Monterey) and XCode 13.3 (13E113)
